File: lab/sortCombined.py
def sort_combine():
    """Sorting combined two sorted arrays"""
    s1 = input_string()
    s2 = input_string()
    num_1 = con_string(s1)
    num_2 = con_string(s2)
    combined_num = combine(num_1, num_2)
    s = con_list(combined_num)
    print(s)


def con_list(num):
    """Converting the list to the string of integer"""
    s_list = [str(i) for i in num]
    s = ','.join(s_list)
    return s


def combine(num_1, num_2):
    """Combined and Sorted"""
    len_1 = len(num_1)
    len_2 = len(num_2)
    combined_num = [0]*(len_1+len_2)
    temp, temp_1, temp_2 = 0, 0, 0
    while temp_1 < len_1 and temp_2 < len_2:
        if num_1[temp_1] < num_2[temp_2]:
            combined_num[temp] = num_1[temp_1]
            temp_1 += 1
        else:
            combined_num[temp] = num_2[temp_2]
            temp_2 += 1
        temp += 1
    while temp_1 < len_1:
        combined_num[temp] = num_1[temp_1]
        temp += 1
        temp_1 += 1
    while temp_2 < len_2:
        combined_num[temp] = num_2[temp_2]
        temp += 1
        temp_2 += 1
    return combined_num


def con_string(s):
    """Converting the string to the list of integer"""
    s_list = s.split(",")
    num_list = []
    for i in s_list:
        num_list.append(int(i))
    return num_list


def input_string():
    """Input the string of array"""
    s = str(input("Input the string of array: "))
    return s


sort_combine()

File: CS210FZ/lab/sortCombined.java
import java.util.Scanner;

public class sortCombined {
    public static void main(String[] args) {
        // Given an input string of a sequence
        // of integers from the console.
        Scanner in = new Scanner(System.in);

        // Create a String
        String s1 = in.nextLine();
        String s2 = in.nextLine();

        // Change the string to the integer array
        int[] num1 = createArray(s1);
        int[] num2 = createArray(s2);

        // Combine those arrays being one and
        // sorting it from small to large.
        int[] combine =combinedTwoArray(num1, num2);

        for(int i = 0; i < combine.length; i++) {
            if (i == combine.length-1) {
                System.out.println(combine[i]);
                break;
            }
            System.out.print(combine[i]+",");
        }
        in.close();
    }

    // Create an array of integer，
    // read the string and parse
    // it as an unsorted array of integers
    public static int[] createArray(String s){
        // Read the string and parse it
        // as an unsorted array of integers
        String[] stringNum = s.split(",");
        int len = stringNum.length;
        int[] num = new int[len];
        for (int i = 0; i < len; i++){
            num[i] = Integer.parseInt(stringNum[i]);
        }
        return num;
    }

    public static int[] combinedTwoArray(int[] num1, int[] num2) {
        int len1 = num1.length,
                len2 = num2.length,
                len = len1 + len2;
        int temp1 = 0, temp2 = 0, temp = 0;
        int[] combine = new int[len];
        while(temp1 < len1 && temp2 < len2){
            combine[temp++] = num1[temp1] < num2[temp2] ? num1[temp1++] : num2[temp2++];
        }
        while(temp1 < len1){
            combine[temp++] = num1[temp1++];
        }
        while(temp2 < len2){
            combine[temp++] = num2[temp2++];
        }
        return combine;
    }
}
